pyinfraformat/core/io.py

```diff
@@ -136,68 +136,66 @@ def from_gtk_wfs(bbox, coord_system, robust=True, maxholes=1000):
     holes = from_gtk_wfs(bbox, coord_system="EPSG:4326", robust=True)
     """
     # pylint: disable=invalid-name
-    epsg_names = {EPSG_SYSTEMS[i]: i for i in EPSG_SYSTEMS}
     url = "http://gtkdata.gtk.fi/arcgis/services/Rajapinnat/GTK_Pohjatutkimukset_WFS/MapServer/WFSServer?"  # pylint: disable=line-too-long
-    wfs = WebFeatureService(url)
+    wfs = WebFeatureService(url, version="2.0.0")
 
-    x1, y1 = project_points(bbox[0], bbox[1], coord_system, "EPSG:3067")
-    x2, y2 = project_points(bbox[2], bbox[3], coord_system, "EPSG:3067")
+    x1, y1 = project_points(bbox[0], bbox[1], coord_system, "EPSG:4326")
+    x2, y2 = project_points(bbox[2], bbox[3], coord_system, "EPSG:4326")
 
     x1, x2 = min((x1, x2)), max((x1, x2))
     y1, y2 = min((y1, y2)), max((y1, y2))
-    bbox = [y1, x1, y2, x2]
+    bbox = [x1, y1, x2, y2]
 
     wfs_io = wfs.getfeature(
-        typename=["Pohjatutkimukset"], maxfeatures=maxholes, srsname="EPSG:3067", bbox=bbox
+        typename=["Rajapinnat_GTK_Pohjatutkimukset_WFS:Pohjatutkimukset"],
+        maxfeatures=maxholes,
+        bbox=bbox,
+        outputFormat="GEOJSON",
     )
 
-    data = wfs_io.read()
-    data_dict = xmltodict.parse(data)
-
-    if "gml:featureMember" not in data_dict["wfs:FeatureCollection"]:
+    data = wfs_io.read().decode("utf-8")
+    data = data.replace("\\", r"\\")
+    data_json = json.loads(data, strict=False)
+    if "features" not in data_json:
         return Holes()
 
     results = []
-    for i in data_dict["wfs:FeatureCollection"]["gml:featureMember"]:
-        line = dict(
-            **i["Rajapinnat_GTK_Pohjatutkimukset_WFS:Pohjatutkimukset"][
-                "Rajapinnat_GTK_Pohjatutkimukset_WFS:Shape"
-            ]["gml:Point"],
-            **i["Rajapinnat_GTK_Pohjatutkimukset_WFS:Pohjatutkimukset"]
-        )
+    for line in data_json["features"]:
         results.append(line)
-
     holes = []
 
     def parse_line(line):
-        if "Rajapinnat_GTK_Pohjatutkimukset_WFS:ALKUPERAINEN_DATA" in line:
-            hole_str = line["Rajapinnat_GTK_Pohjatutkimukset_WFS:ALKUPERAINEN_DATA"].split("\n")
+        if ("properties" in line) and ("ALKUPERAINEN_DATA" in line["properties"]):
+            hole_str = line["properties"]["ALKUPERAINEN_DATA"].split("\r")
             hole = parse_hole(enumerate(hole_str), robust=robust)
         else:
             hole = Hole()
+        hole.add_header("OM", {"Owner": line["properties"]["OMISTAJA"]})
+
+        x, y = line["geometry"]["coordinates"]
```

Comment:
geojson should always have geometry, right?

Reply:
Yea, and probably we want such an error to raise an error; where would that data come from?

Reply:
Sounds reasonable